lib/create-file.js

```diff
@@ -1,27 +1,25 @@
 'use strict';
 
 const templateFactory = require('./template-factory');
-const { writeFile } = require('../lib/lib-helpers');
+const { writeFile } = require('../lib/helpers');
 const path = require('path');
 
-const createComponent = (name, isDumb, createDir, overwriteFile, options) => {
-    let dirPath = path.normalize(path.resolve(name));
-    let file = `${dirPath}.js`;
-    let compName = path.basename(name);
-    let template = isDumb
-        ? templateFactory('DUMB_COMPONENT', compName)
-        : templateFactory('COMPONENT', compName, options);
+const createComponent = (fileName, filePath, options, configObj) => {
+    let fileObj = path.parse(filePath);
+    filePath = fileObj.ext ? filePath : `${filePath}.js`;
+    let template = options.dumb
+        ? templateFactory('DUMB_COMPONENT', fileName)
+        : templateFactory('COMPONENT', fileName, configObj);
 
-    writeFile(file, template, overwriteFile, createDir);
+    writeFile(filePath, template, options.overwrite, options.createDir);
 };
 
-const createTest = (name, overwriteFile) => {
-    let dirPath = path.normalize(path.resolve(name));
-    let file = `${dirPath}.test.js`;
-    let compName = path.basename(name);
-    let template = templateFactory('COMPONENT_TEST', compName);
+const createTest = (fileName, filePath, options, configObj, configStore) => {
+    let dirPath = path.dirname(filePath);
+    filePath = path.resolve(dirPath, `${fileName}.test.js`);
+    let template = templateFactory('COMPONENT_TEST', fileName, configStore);
 
-    writeFile(file, template, overwriteFile);
+    writeFile(filePath, template, options.overwrite);
 };
```

Comment:
So before we merge this in, trying to think towards the future... we're going to be looking at probably at least 1 or 2 more "file types"... do you think the best approach is to have a function for each type (eg; createComponent, createTest, createReducer, createStore, etc) or should we try to merge it into bigger function to to keep it a little DRYer?

On one hand, I think having a separate function for each makes the code a bit more readable and adhere's more to separation of concern's, but there's also a lot of repetitive code, so it makes it less DRY... What are your thoughts?  Maybe we should find some middle ground and have almost another factory-type setup to determine the template and filename?  I'm really not sure, lol.

Reply:
Yea, after sleeping on it, I think I agree with you.  We may repeat a bit of code, but I think it does keep it much more readable and easier to work on.  Thanks for keeping me honest!

Reply:
Funny thing... After refactoring the formatting of the `filePath` I saw an opportunity to convert `createFIle` into the factory-type function you mentioned. This will definitely make things cleaner while still allowing us to add `fileType` specific behavior in the future! Check out the new commit and let me know what you think.